packages/block-library/src/categories/edit.js

```diff
@@ -257,9 +257,11 @@ export default function CategoriesEdit( {
 			) }
 			{ ! isResolving &&
 				categories?.length > 0 &&
-				( displayAsDropdown
-					? renderCategoryDropdown()
-					: renderCategoryList() ) }
+				( displayAsDropdown ? (
+					renderCategoryDropdown()
+				) : (
+					<nav>{ renderCategoryList() }</nav>
+				) ) }
```

Comment:
By placing the `<nav>` inside `<TagName>` the resulting markup is a `<nav>` inside a `<ul>`, which is not valid HTML.

Instead, the `TagName` constant should be updated and the block props should be placed on the new outer element, the `nav.`

```
	const TagName =
		!! categories?.length && ! displayAsDropdown && ! isResolving
			? 'ul'
			: 'div';

```